src/glue/wasm/i64.ts

```diff
@@ -68,176 +68,165 @@ function i64_pow(left: i64, right: i64): i64 {
+@global @inline
 function i64_shr_u(left: i64, right: i64): i64 {
   return left >>> right;
 }
 
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_not(value: i64): i64 {
   return ~value;
 }
 
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_eq(left: i64, right: i64): bool {
   return left == right;
 }
 
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_ne(left: i64, right: i64): bool {
   return left != right;
 }
 
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_gt(left: i64, right: i64): bool {
   return left > right;
 }
 
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_align(value: i64, alignment: i64): i64 {
   var mask: i64 = alignment - 1;
   assert(alignment && (alignment & mask) == 0);
   return (value + mask) & ~mask;
 }
 
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_is_i8(value: i64): bool {
   return value >= i8.MIN_VALUE && value <= <i64>i8.MAX_VALUE;
 }
 
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_is_i16(value: i64): bool {
   return value >= i16.MIN_VALUE && value <= <i64>i16.MAX_VALUE;
 }
 
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_is_i32(value: i64): bool {
   return value >= i32.MIN_VALUE && value <= <i64>i32.MAX_VALUE;
 }
 
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_is_u8(value: i64): bool {
   return value >= 0 && value <= <i64>u8.MAX_VALUE;
 }
 
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_is_u16(value: i64): bool {
   return value >= 0 && value <= <i64>u16.MAX_VALUE;
 }
 
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_is_u32(value: i64): bool {
   return value >= 0 && value <= <i64>u32.MAX_VALUE;
 }
 
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_is_bool(value: i64): bool {
   return value === 0 || value === 1;
 }
 
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_is_f32(value: i64): bool {
   return value >= <i64>f32.MIN_SAFE_INTEGER && value <= <i64>f32.MAX_SAFE_INTEGER;
 }
 
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_is_f64(value: i64): bool {
   return value >= <i64>f64.MIN_SAFE_INTEGER && value <= <i64>f64.MAX_SAFE_INTEGER;
 }
 
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_to_f32(value: i64): f32 {
   return <f32>value;
 }
 
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_to_f64(value: i64): f64 {
   return <f64>value;
 }
 
-import { CharCode } from "../../util";
-
 // @ts-ignore: decorator
-@global
+@global @inline
 function i64_to_string(value: i64, unsigned: bool = false): string {
-  var chars = new Array<i32>();
-  if (!unsigned && value < 0) {
-    chars.push(CharCode.MINUS);
-    value = -value;
-  }
-  do {
-    chars.push(CharCode._0 + <i32>(<u64>value % 10));
-    value = <u64>value / 10;
-  } while (value);
-  return String.fromCharCodes(chars);
+  return (unsigned ? <u64>value : value).toString();
```

Comment:
Looks like the type of the conditional is always u64 here, in turn always calling U64#toString.

Reply:
Good catch!